src/interfaces/controllers/get-trailer-registration.ts

```diff
@@ -0,0 +1,30 @@
+import { Request, Response, NextFunction } from 'express';
+import * as domain from '../../domain';
+import { getTrailerValidator } from '../validators/get-trailer-registration';
+import { ERRORS, GetTrailerRequest } from '../../domain';
+import { TrailerRegistrationBase } from './trailer-registration-base';
+
+export class GetTrailerRegistration extends TrailerRegistrationBase {
+  private validate(payload: GetTrailerRequest): string {
+    const errors = getTrailerValidator.validate(payload).error;
+    return errors?.message;
+  }
+
+  public async call(req: Request, res: Response, next: NextFunction): Promise<void> {
+    const { vin } = req.params;
+    const { make } = req.query;
+    const payload: GetTrailerRequest = { vin, make: make as string };
+    const errors = this.validate(payload);
+    if (errors) {
+      next(new domain.HTTPError(400, errors));
+      return;
+    }
+    const vinOrChassisWithMake = vin.length === 17 ? vin : vin + payload.make;
```

Comment:
We've got this logic in a few places. Would it be better to externalise it to some function that given a vin and a make returns whatever the vinOrChassisWithmake should be?